docs/source/validation_benchmarking/ipf_performance.rst

```diff
@@ -1,39 +1,62 @@
+than the benchmark python code, while applying IPF to a 32-bit NumPy array (np.float32) was significantly faster.
+It's worth mentioning that
 the user can set up a threshold for AequilibraE's IPF function, as well as use more than one
 core to speed up the fitting process.
 
 .. image:: ../images/ipf_runtime_aequilibrae_vs_benchmark.png
     :align: center
     :alt: AequilibraE's IPF runtime
 
-As IPF is an embarrassingly-parallel workload, we have implemented our version in Cython, taking full advantage
-of parallelization and observing the impact of array orientation in memory. AequilibraE's
-IPF allows the user to choose how many cores are used for IPF in order to speed up the fitting process, which
-is extremely useful when handling models with lots of traffic zones.
+As IPF is an embarrassingly-parallel workload, it is more relevant to look at the performance of the
+AequilibraE implementations, starting by comparing the implementation performance for inputs in 32 vs 64
+bits using 32 threads.
+
+.. image:: ../images/ipf_runtime_32vs64bits.png
+    :align: center
+    :alt: AequilibraE's IPF runtime 32 vs 64 bits
+
+The difference is staggering, with the 32-bit implementation being twice as fast as the 64-bit one for large matrices.
+It is also worth noting that differences in results between the outputs between these two versions are incredibly
+small (RMSE < 1.1e-10), and therefore unlikely to be relevant in most applications.
 
-As we can see, instances with more zones benefited the most from the power of multi-processing
-speeding up the runtime in barely five times using five cores.
+We can also look at performance gain across matrix sizes and number of cores, and it becomes clear
+that the 32-bit version scales significantly better than its 64-bit counterpart, showing significant performance
+gains up to 16 threads, while the latter stops showing much improvement beyond 8 threads, likely due to limitations
+on cache size.
 
 .. image:: ../images/ipf_runtime_vs_num_cores.png
-    :align: center
-    :alt: number of cores used in IPF
+    :align: left
+    :alt: number of cores used in IPF for 64 bit matrices
+
+.. image:: ../images/ipf_runtime_vs_num_cores32bits.png
+    :align: right
+    :alt: number of cores used in IPF for 32 bit matrices
+
+In conclusion, AequilibraE's IPF implementation is over 11 times faster than its pure Python counterpart for
+large matrices on a workstation, largely due to the use of Cython and multi-threading, but also due to the use of a
+32-bit version of the algorithm.
 
 These tests were run on a Threadripper 3970x (released in 2019) workstation with 32 cores (64 threads) @ 3.7 GHz
-and 256 Gb of RAM. With 32 cores in use, performing IPF took 0.105s on a 10,000 zones matrix,
-and 0.224 seconds on a 15,000 matrix. The code is provided below for convenience
+and 256 Gb of RAM. The code is provided below for reference.
 
 .. _code-block-for-ipf-benchmarking:
```

Comment:
@r-akemii , can you replace the code below with a rendered version of IPF_benchmark.ipynb here?

Reply:
@pedrocamargo I've added the rendered notebook as part of the documentation, but I kept this ipf_performance page because latexpdf is having a bad time with an unicode character from the render.